beamer/agent/chain.py

```diff
@@ -601,8 +601,12 @@ def maybe_prove(claim: Claim, context: Context) -> None:
     elif claim.invalidation_tx is not None:
         prove_tx = claim.invalidation_tx
 
-    # mainnet: 10, goerli: 420, local: 901
-    if context.target_chain.id not in (10, 420, 901):
+    chain_descriptor = get_chain_descriptor(context.target_chain.id)
+
+    if chain_descriptor is None:
+        return
```

Comment:
At this point it is guaranteed that `chain_descriptor` cannot be `None`, right?
If it was `None`, that means that a request for an unsupported chain somehow got through our machinery to `maybe_prove`, but I was convinced that we filter out such requests/events much earlier, when we get them.
I would appreciate if you could check that and turn this check rather into an `assert` statement.